codalab/lib/upload_manager.py

```diff
@@ -168,14 +126,10 @@ def _simplify_archive(self, path):
         if not os.path.isdir(path):
             return
 
-        files = [f for f in os.listdir(path) if not self._ignore_file_in_archive(f)]
+        files = [f for f in os.listdir(path)]
```

Comment:
I went ahead and removed this logic, as well. Although it's technically a breaking change in how CodaLab treats certain files, I don't think we want to keep it because the current behavior is problematic.

Previous behavior (current behavior on master):
- If you upload an archive to CodaLab with exactly two files: `README` and `.DS_Store`, CodaLab will just treat the bundle as a single file called `README`.

New behavior (with this PR):
- If you upload an archive to CodaLab with exactly two files: `README` and `.DS_Store`, CodaLab will treat the bundle as a folder with two files: `README` and `.DS_Store`.

The previous behavior is problematic because the user **has already included both files** in the fileobj they are uploading. Presumably, if they really wanted to ignore `.DS_Store`, they would have done so with `cl upload --exclude-patterns`, and the fileobj wouldn't have included `.DS_Store` in the first place.

Because fileobj _does_ include both these files, the new behavior in this PR makes the most sense.

I removed this logic now so I wouldn't have to replicate it for Blob Storage (thus simplifying blob storage implementation).

This scenario is also covered in the `test_fileobj_single_tar_gz_with_dsstore_should_not_simplify_archive` test case.